File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/_base/BaseModuleViewModel.java
package amagi82.modularcharactersheetcreator.ui._base;

import amagi82.modularcharactersheetcreator.models.modules.Module;

public abstract class BaseModuleViewModel {
    @Module.Type public final int type;
    @Module.SpanCount public final int spanCount;

    public BaseModuleViewModel(Module module) {
        type = module.type();
        spanCount = module.spanCount();
    }

    public abstract void update(Module module);
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/_base/BaseModuleViewModel.kt
package amagi82.modularcharactersheetcreator.ui._base

import amagi82.modularcharactersheetcreator.extras.ModuleSpanCount
import amagi82.modularcharactersheetcreator.extras.ModuleType
import amagi82.modularcharactersheetcreator.models.modules.Module

abstract class BaseModuleViewModel(module: Module) {
    @ModuleType val type: Int
    @ModuleSpanCount val spanCount: Int

    init {
        type = module.type
        spanCount = module.spanCount
    }

    abstract fun update(module: Module)
}
